client/files.go

```diff
@@ -0,0 +1,226 @@
+package client
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+
+	"github.com/beyondstorage/go-storage/v4/pairs"
+	"github.com/beyondstorage/go-storage/v4/services"
+	"github.com/beyondstorage/go-storage/v4/types"
+
+	"github.com/beyondstorage/beyond-ftp/utils"
+)
+
+func (c *Handler) handleSTOR() {
+	path := c.absPath(c.param)
+
+	c.storeOrAppend(path, false)
+}
+
+func (c *Handler) handleAPPE() {
+	_, ok := c.storager.(types.Appender)
+	if !ok {
+		c.WriteMessage(StatusCommandNotImplemented, "this type of storage is not support append")
+		return
+	}
+	path := c.absPath(c.param)
+	c.storeOrAppend(path, true)
+}
+
+func (c *Handler) storeOrAppend(path string, append bool) {
+	c.ctxRest = 0
+	tr, err := c.TransferOpen()
+	if err != nil {
+		c.WriteMessage(StatusCannotOpenDataConnection, err.Error())
+		return
+	}
+
+	if err := c.upload(path, tr, append); err != nil {
+		c.TransferClose()
+		c.WriteMessage(StatusFileActionNotTaken, err.Error())
+		return
+	}
+
+	select {
+	case <-c.commandAbortCtx.Done():
+		c.WriteMessage(StatusTransferAborted, "Connection closed; transfer aborted")
+	default:
+		c.TransferClose()
+		c.WriteMessage(StatusClosingDataConn, "transfer finished")
+	}
+}
+
+func (c *Handler) upload(path string, tr utils.Conn, append bool) error {
```

Comment:
We should check validation first, and then process IO, because IO cost too much. It is acceptable to repeat `io.Copy` when coding.